plugins/modules/ec2_transit_gateway_vpc_attachment_info.py

```diff
@@ -162,39 +176,45 @@ def main():
     module = AnsibleAWSModule(
         argument_spec=argument_spec,
         supports_check_mode=True,
+        mutually_exclusive=mutually_exclusive,
     )
 
-    name = module.params.get("name", None)
-    id = module.params.get("id", None)
-    opt_filters = module.params.get("filters", None)
+    name = module.params.get("name")
+    attachment_id = module.params.get("id")
+    opt_filters = module.params.get("filters")
+    include_deleted = module.params.get("include_deleted")
+
+    client = module.client("ec2")
+
+    params: Dict[str, Any] = {}
+    filters: Dict[str, Any] = {}
+    attachments: List[Dict[str, Any]] = []
 
-    search_manager = TransitGatewayVpcAttachmentManager(module=module)
-    filters = dict()
+    if attachment_id:
+        params["TransitGatewayAttachmentIds"] = [attachment_id]
 
+    # Add filter by name if provided
     if name:
         filters["tag:Name"] = name
 
-    if not module.params.get("include_deleted"):
-        # Attachments lurk in a 'deleted' state, for a while, ignore them so we
-        # can reuse the names
-        filters["state"] = [
-            "available",
-            "deleting",
-            "failed",
-            "failing",
-            "initiatingRequest",
-            "modifying",
-            "pendingAcceptance",
-            "pending",
-            "rollingBack",
-            "rejected",
-            "rejecting",
-        ]
+    # Include only active states if "include_deleted" is False
+    if not include_deleted:
+        filters["state"] = get_states()
 
+    # Include any additional filters provided by the user
     if opt_filters:
         filters.update(opt_filters)
 
-    attachments = search_manager.list(filters=filters, id=id)
+    if filters:
+        params["Filters"] = ansible_dict_to_boto3_filter_list(filters)
+
+    try:
+        result = describe_vpc_attachments(client, **params)
+    except AnsibleEC2Error as e:
+        module.fail_json_aws_error(e)
+
+    if result:
+        attachments = [boto3_resource_to_ansible_dict(attachment) for attachment in result]
 
     module.exit_json(changed=False, attachments=attachments, filters=filters)
```

Comment:
This is what module used to return. We cannot remove it because it will be a breaking change. After module's migration to amazon.aws we can deprecate this information.

Reply:
you should then update the RETURN block accordingly